src/appleseed.studio/mainwindow/mainwindow.cpp

```diff
@@ -181,6 +183,16 @@ void MainWindow::open_project(const QString& filepath)
     m_project_manager.load_project(filepath.toAscii().constData());
 }
 
+void MainWindow::new_project()
+{
+    if (!can_close_project())
```

Comment:
I think that in scripting, we always want to close existing project, even if it has unsaved changes.
We should move the can_close_project() test back to the slot.
What do you think?